awslabs/amazon-kinesis-video-streams-webrtc-sdk-js
# Sample: Master Cleanup

## Description of changes
* Move common code that can be shared between master and viewer to a separate class w/ appropriate public interfaces. This makes the AWS SDK calls abstracted out and make the sample more readable and easy to modify.

## Testing

Manual testing:
* Local changes (master, chrome) --> Deployed JS Webpage (viewer, chrome)
* Local changes (master, firefox) --> Deployed JS Webpage (viewer, chrome)
* Local changes (master, safari) --> Deployed JS Webpage (viewer, chrome)
* Local changes (master, chrome) --> Deployed JS Webpage (viewer, chrome) sending/receiving datachannel messages on both sides
* Local changes (master, chrome) --> iOS (viewer)
* Local changes (master, chrome) --> Android (viewer)
* Local changes (master, chrome) --> WebRTC ingestion [success case]
* Local changes (master, chrome) --> WebRTC ingestion [failure case, channel not configured for ingestion, verify that application exits]
* Local changes (master, chrome) --> WebRTC ingestion [failure case, verify the 5 connections within 10 minutes]
* Local changes (master, firefox) --> WebRTC ingestion [failure case, verify that statusResponse is still printed and application exits]

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
